packages/core/src/utility.ts

```diff
@@ -79,3 +79,10 @@ export const readVarInt = (buffer: Buffer, offset: number = 0): number => {
 export const encodingLength = (n: number) => {
     return n < 0xfd ? 1 : n <= 0xffff ? 3 : n <= 0xffffffff ? 5 : 9;
 };
+
+export const isPubKey = (testVector: Buffer | null): boolean => {
+    if (testVector && testVector.length == 33) {
```

Comment:
here we should also verify the public key using `secp256k1` to confirm if it's an actual public key and not some 33 byte random string